src/ui/public/vislib/visualizations/pie_chart.js

```diff
@@ -322,6 +323,61 @@ export function VislibVisualizationsPieChartProvider(Private) {
       }
     }
 
+    readableNumber(number) {
+      let tier = Math.log10(Math.abs(number)) / 3 | 0;
+      if(tier === 0) return number;
+      if(tier > 4) tier = 4;
+      const postfix = this.ABBREVIATIONS[tier];
+      const scale = Math.pow(10, tier * 3);
+      const scaled = number / scale;
+      let formatted = String(scaled.toFixed(1));
+      if (/\.0$/.test(formatted)) {
```

Comment:
We can get rid of the whole (expensive) regex operations, string cutting and even using `let`, by doing the following instead:

```js
// Converting a string to a numeric value will cut off any unnecessary trailing zeroes
const formatted = Number(scaled.toFixed(1));
return `${formatted}${postfix}`;
```